src/Core/Scheduler/Cache.js

```diff
@@ -0,0 +1,146 @@
+     *
+     * @name module:Cache
+     * @typedef {Object} module:Cache.POLICIES
+     *
+     * @property INFINITE - The entry is never flushed, except when the
+     * <code>all</code> flag is set to <code>true</code> when calling {@link
+     * Cache.flush}.
+     * @property TEXTURE - Shortcut for texture resources. Time is 15 minutes.
+     * @property ELEVATION - Shortcut for elevation resources. Time is 15
+     * minutes.
+     */
+    POLICIES: {
+        INFINITE: Infinity,
+        TEXTURE: 900000,
+        ELEVATION: 900000,
+    },
+
+    /**
+     * Returns a boolean indicating whether an entry with the specified key
+     * exists in the cache.
+     *
+     * @name module:Cache.has
+     * @function
+     *
+     * @param {string} key
+     *
+     * @return {boolean}
+     */
+    has: key => data.has(key),
+
+    /**
+     * Returns the entry related to the specified key from the cache. The last
+     * time used property of the entry is updated to extend the longevity of the
+     * entry.
+     *
+     * @name module:Cache.get
+     * @function
+     *
+     * @param {string} key
+     *
+     * @return {Object}
+     */
+    get: (key) => {
+        const entry = data.get(key);
+
+        if (entry) {
+            entry.lastTimeUsed = Date.now();
+            return entry.value;
+        }
+    },
+
+    /**
+     * Adds or updates an entry with a specified key. A lifetime can be added,
+     * by specifying a numerical value or using the {@link Cache.POLICIES}
+     * values. By default an entry has an infinite lifetime.
+     *
+     * @name module:Cache.set
+     * @function
+     *
+     * @param {string} key
+     * @param {Object} value
+     * @param {number} [lifetime]
+     *
+     * @return {Object} the added value
+     */
+    set: (key, value, lifetime = Infinity) => {
+        const entry = {
+            value,
+            lastTimeUsed: Date.now(),
+            lifetime,
+        };
+        data.set(key, entry);
+
+        return value;
+    },
+
+    /**
+     * Deletes the specified entry from the cache.
+     *
+     * @name module:Cache.delete
+     * @function
+     *
+     * @param {string} key
+     *
+     * @return {boolean} - Confirmation that the entry has been deleted.
+     */
+    delete: key => data.delete(key),
+
+    /**
+     * Removes all entries of the cache.
+     *
+     * @name module:Cache.clear
+     * @function
+     */
+    clear: data.clear(),
+
+    /**
+     * Flush the cache: entries that have been present for too long since the
+     * last time they were used, are removed from the cache. By default, the
+     * time is the current time, but the interval can be reduced by doing
+     * something like <code>Cache.flush(Date.now() - reductionTime)</code>. If
+     * you want to clear the whole cache, use {@link Cache.clear} instead.
+     *
+     * @name module:Cache.flush
+     * @function
+     *
+     * @param {number} [time]
+     */
+    flush: (time = Date.now()) => {
```

Comment:
Having a cache is good, but having statistics on how much hit/miss it generates is better.

Could you modify `flush` to return something easy to log/plot: `(hit, miss, entry_count)` for instance?
Or even better: add a new chart that draws these so we can see how useful this cache is...

Reply:
Added, but keep in mind that the current time values are not big enough to see rapid changes.